cl/citations/tests.py

```diff
@@ -212,46 +210,42 @@ def test_make_html_from_matched_citation_objects(self) -> None:
-             'MATCH_ID">, <a href="MATCH_URL"><span class="id_token">Id.,'
-             '</span> at 123, 124</a></span><pre class="inline">. Lorem ipsum'
-             ' dolor sit amet</pre>'),
+             '<pre class="inline">asdf, </pre><span class="citation" data-id="'
+             'MATCH_ID"><a href="MATCH_URL">Id., at 123, 124.</a></span><pre '
+             'class="inline"> Lorem ipsum dolor sit amet</pre>'),
 
             # Id. citation with complex page number ("Id. @ 123:1, ¶¶ 124")
             ('asdf, Id. @ 123:1, ¶¶ 124. Lorem ipsum dolor sit amet',
-             IdCitation(id_token='Id.',
-                        after_tokens=['@', '123:1', '¶¶', '124'],
-                        has_page=True),
-             '<pre class="inline">asdf</pre><span class="citation" data-id="'
-             'MATCH_ID">, <a href="MATCH_URL"><span class="id_token">Id.'
-             '</span> @ 123:1, ¶¶ 124</a></span><pre class="inline">. Lorem '
-             'ipsum dolor sit amet</pre>'),
+             '<pre class="inline">asdf, </pre><span class="citation" data-id='
+             '"MATCH_ID"><a href="MATCH_URL">Id. @ 123:1, ¶¶ 124.</a></span>'
+             '<pre class="inline"> Lorem ipsum dolor sit amet</pre>'),
 
             # Id. citation without page number ("Id. Something else")
             ('asdf, Id. Lorem ipsum dolor sit amet',
-             IdCitation(id_token='Id.', after_tokens=['Lorem', 'ipsum'],
-                        has_page=False),
-             '<pre class="inline">asdf</pre><span class="citation" data-id='
-             '"MATCH_ID">, <a href="MATCH_URL"><span class="id_token">Id.'
-             '</span></a> Lorem ipsum </span><pre class="inline">dolor sit '
-             'amet</pre>'),
+             '<pre class="inline">asdf, </pre><span class="citation" data-id="'
+             'MATCH_ID"><a href="MATCH_URL">Id.</a></span><pre class="inline">'
+             ' Lorem ipsum dolor sit amet</pre>'),
         ]
 
         # fmt: on
-        for plain_text, citation, expected_html in test_triples:
+        for s, expected_html in test_pairs:
             print(
-                "Testing object to HTML rendering for %s..." % plain_text,
+                "Testing object to HTML rendering for %s..." % s,
                 end=" ",
             )
-            citation.match_url = "MATCH_URL"
-            citation.match_id = "MATCH_ID"
-            opinion = Opinion(plain_text=plain_text)
-            created_html = create_cited_html(opinion, [citation])
+            opinion = Opinion(plain_text=s)
+            get_and_clean_opinion_text(opinion)
+            citations = get_citations(opinion.cleaned_text)
+            for c in citations:  # Fake correct matching for the citations
+                c.match_url = "MATCH_URL"
+                c.match_id = "MATCH_ID"
+            created_html = create_cited_html(
+                opinion=opinion,
+                citations=citations,
+            )
             self.assertEqual(
```

Comment:
It'd be nice to use self.subTest for any of these with lists of test cases. Makes them a lot easier to deal with.

Reply:
Interesting trick, I will do that.

Reply:
Got it from Jack in eyecite. It's new in py3.